woocommerce/woocommerce-ios
# Notifications: improve Dynamic Type support in the body of the notification

Fixes #912  

## Changes
The fixes in this PR are for two main issues:
- The font size of the subject and snippet labels did not update dynamically, but only on app launch:
  -  I haven't found concrete/official evidence online to support my fix, but I believe it's caused by using a static constant `NSAttributedString` styles dictionary with a fixed [`preferredFont(forTextStyle:)`](https://developer.apple.com/documentation/uikit/uifont/1619030-preferredfont) value for all `NSAttributedString` usage. From all the official and popular examples using `preferredFont(forTextStyle:)` for dynamic type, it is always called on `font` assignment.Open to any other thoughts!
- The notification icon on the left is clipped with larger font sizes. I fixed this one by:
  - Setting `horizontalCompressionResistancePriority` to `required` so that the icon maintains its content size with the highest priority
  - Creating a `UIFont` helper to return a `Noticons` font that supports dynamic type based on a font size, following [Apple documentation](https://developer.apple.com/documentation/uikit/uifont/scaling_fonts_automatically) on custom font. I originally wanted to update the [existing `Noticons` font helper](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/Extensions/UIFont+Helpers.swift#L59-L63), which calculates the font size at the given device font size and returns a `UIFont` with that fixed calculated font size. Because the existing helper works differently (the return `UIFont` does not scale with dynamic type) and is used in other cases, I decided to create a separate helper
  - Notes: I kept the text style for the icon as `title1`, but I also tried `body` and it looks slightly more aligned with the subject with larger font sizes (as shown in screenshots). The icon base font size can also be tweaked - see if @kyleaparker has any preferences. --> Updated to `body` font style

## Testing
- Launch the app with a store that has at least one notification
- Go to "Notifications" tab
- Turn on dynamic type (device Settings > General > Accessibility > Larger Text > turn it on) and adjust the device to different font sizes via the slider, and observe the notification cell --> The labels should adjust their font size dynamically

## Example screenshots

Notes: the notification content and icon color are mocked

text style | 3rd smallest | 8th smallest | largest
--- | --- | --- | ---
`body` with 25pt base font size | ![Simulator Screen Shot - iPhone SE - 2019-07-08 at 17 39 52](https://user-images.githubusercontent.com/1945542/60844532-8c7b1e00-a1a7-11e9-9024-e23ac68e63eb.png) | ![Simulator Screen Shot - iPhone SE - 2019-07-08 at 17 40 01](https://user-images.githubusercontent.com/1945542/60844538-91d86880-a1a7-11e9-9c36-dc866aaed21a.png) | ![Simulator Screen Shot - iPhone SE - 2019-07-08 at 17 39 45](https://user-images.githubusercontent.com/1945542/60844542-943ac280-a1a7-11e9-881f-dec5fff22fc1.png)




Update release notes:
